app/src/main/java/org/mozilla/fenix/FenixApplication.kt

```diff
@@ -56,6 +60,27 @@ open class FenixApplication : LocaleAwareApplication() {
             return
         }
 
+        // We need to always initialize Glean and do it early here. Note that we are disabling it
+        // here too (uploadEnabled = false). If needed Glean will be enabled later by the migration
+        // code (if this user used to be a fennec user with the right flags enabled) or by
+        // GleanMetricsService if telemetry is enabled for this user.
+        // It is important that this initialization happens *here* before calling into
+        // setupInMainProcessOnly() which behaves differently for fenix and fennec builds.
+        // Glean needs to be disabled initially because otherwise we may already collect telemetry
+        // before we know whether we want that (which is *after* the migration).
+        // As a side effect this means pings submitted between the initialization here and until we
+        // potentially enable Glean would be lost. However such pings do not exist at this moment.
+        logger.debug("Initializing Glean (uploadEnabled=false)")
+        Glean.initialize(
+            applicationContext = this,
+            configuration = Configuration(
+                channel = BuildConfig.BUILD_TYPE,
+                httpClient = ConceptFetchHttpUploader(
+                    lazy(LazyThreadSafetyMode.NONE) { components.core.client }
+                )),
+            uploadEnabled = false
```

Comment:
I am concerned that this will cause the dropping of any pre-Glean-initialized events, of which there are a non-zero number currently in Fenix.  (I'm building now to get a current number of what they actually are and how many).  But given that playing them back was the source of the recent performance issue, I don't expect that number to be low.

Of course, given this move to initializing here, perhaps that will no longer be the case, but it would be good to confirm.  Passing in `false` here unconditionally sort of removes one of Glean's completeness guarantees.

Is there any way to get the migrated value for this from Fennec prior to this?